src/lib/dnssd/ServiceNaming.cpp

```diff
@@ -135,6 +135,10 @@ CHIP_ERROR MakeServiceSubtype(char * buffer, size_t bufferLen, DiscoveryFilter s
         break;
     case DiscoveryFilterType::kCompressedFabricId:
         requiredSize = snprintf(buffer, bufferLen, "_I");
+        if (requiredSize < 0)
+        {
+            return CHIP_ERROR_NO_MEMORY;
+        }
```

Comment:
```suggestion
        VerifyOrReturnError((requiredSize < 0) && (requiredSize < bufferLen), CHIP_ERROR_NO_MEMORY);
```

but am I reading this correctly ... is this just figuring out how much buffer space is needed to place a "_I" in the buffer? Why are we not having a constant and strcpy? I guess this was placed based on other places having `%` and doing real work, but this seems bad. I would rather maybe:


```suggestion
        constexpr size_t kRequiredSize = 2;
        VerifyOrReturnError(bufferLen > kRequiredSize, CHIP_ERROR_NO_MEMORY);
        strcpy(buffer, "_I"); 
```

or something like that.